tests/core_tests/chaingen.h

```diff
@@ -981,25 +990,41 @@ inline bool do_replay_file(const std::string& filename)
 
 #define SET_EVENT_VISITOR_SETT(VEC_EVENTS, SETT) VEC_EVENTS.push_back(event_visitor_settings(SETT));
 
-#define GENERATE(filename, genclass) \
-    { \
-        std::vector<test_event_entry> events; \
-        genclass g; \
-        g.generate(events); \
-        if (!tools::serialize_obj_to_file(events, filename)) \
-        { \
-            MERROR("Failed to serialize data to file: " << filename); \
-            throw std::runtime_error("Failed to serialize data to file"); \
-        } \
+#define GENERATE(filename, genclass)                                                                       \
+  if (epee::file_io_utils::is_file_exist(filename))                                                        \
+  {                                                                                                        \
+    MGINFO_GREEN("Already generated file: " << filename);                                                  \
+  }                                                                                                        \
```

Comment:
I meant to comment on that part.

Reply:
> How does this tell when the cache should be regenerated ?

Unfortunately this is the missing part. Handling this in a smart way would overbloat the PR, which is already large.
What I can offer right now, is setting the CMake variable `TESTS_REUSE_CACHE` to `ON` in the Makefile target, so that it's clearer to the caller, that he can set it to `OFF`, if his PR is about working with the regeneration of the data. 
Would that be enough for this PR?

Reply:
So if you're running as a one off, it won't speed up anything, and if you're running many times, presumably because you're changing the code, you have to reconfigure every time you know the generated data changed, or did not change when it had changed the time before ? When does this actually speed things up ?

Reply:
The cache is being created on the first run.
Only when you know, that your changes alter the generated part, you could alter the Makefile to switch the cache to off. 

There is a risk, that somebody generates the original cache, alters the generation with a failty code, reuses the cache and the tests pass.

Therefore I'd propose:
1) leaving the original generation on the fly as the default option (so no caching),
2) allowing the contributor to temporarily change it (cache the data) until the PR is in shape (or just doing this locally),
3) expecting the contributor to change it back to the original form,
4) confirming, that the tests pass with the original in-place generation.

Reply:
It seems much simpler to just use keccak instead of randomx in core_tests. It speeds up massively and avoids issues of bad cache (that's what I use). Then again, if the default is to not cache, I'm ok with this patch too.

Reply:
> It seems much simpler to just use keccak instead of randomx in core_tests. It speeds up massively and avoids issues of bad cache (that's what I use). Then again, if the default is to not cache, I'm ok with this patch too.

An interesting idea to explore.

I set the caching to OFF by default.